presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -3030,6 +3034,19 @@ public void commit()
         metastore.commit();
     }
 
+    @Override
+    public PartitioningFilterCoverage getPartitioningFilterCoverage(ConnectorTableLayoutHandle connectorTableLayoutHandle, Set<String> relevantPartitionColumns)
+    {
+        HiveTableLayoutHandle tableHandle = (HiveTableLayoutHandle) connectorTableLayoutHandle;
+        List<String> partitionKey = tableHandle.getPartitionColumns().stream().map(HiveColumnHandle::getName).collect(toList());
+        Set<String> relevantColumns = partitionKey.stream().filter(relevantPartitionColumns::contains).collect(toSet());
+        if (relevantColumns.isEmpty()) {
+            return UNKNOWN;
```

Comment:
hmm... this basically says "the table is not partitioned; or it's not partitioned on known columns (e.g. `ds`)". 

Returning `UNKNOWN` for this is a bit weird. Because we know there is no violation / concern for sure. But I also don't have a good suggestions. 

See if @rschlussel has any thought about how the enum values in `PartitioningFilterCoverage` should be . Otherwise, I am fine to keep it as is :)

Reply:
Unknown is pulling double duty as both unknown (not implemented by the connector) and not applicable (we know it's fine).

We could add a fourth value, the question is does that make the API easier to consume when there is nothing that leverages the distinction between unknown and not applicable.